### ISSUE TITLE
Generate tests in forEach loop mark first loop with tests as skipped 

### ISSUE THREAD

**Issue by PeaceDate** [Shelex/cypress-allure-plugin#188]
Reopened #183 

Updated to latest version of Cypress allure plugin. Problem still reproduce

Steps to reproduce
1. Clone this [repo](https://github.com/PeaceDate/cypress-allure-plugin-bug)
2. Run `yarn install`
3. Run `npm run cypress:test`
4. Run `npm run report:generate`
5. Run `npm run report:open`

Result -> First test ignored and marked as skipped
<img width="1726" alt="Screenshot 2023-02-02 at 22 06 09" src="https://user-images.githubusercontent.com/55971575/216438109-464d8b20-5c29-4833-84f5-e6c0db5f392c.png">

@shelex/cypress-allure-plugin: "2.35.0"
cypress: "10"

**Comment by Shelex** [comment]
@PeaceDate Thank you for an example repo, that helped a lot.

Now I see that problem is not a forEach loop actually, but the changing domain, so this issue is identical to https://github.com/Shelex/cypress-allure-plugin/issues/179 and https://github.com/Shelex/cypress-allure-plugin/issues/125. 

It is caused by a way Cypress work - `baseUrl` is absent, same file involving different domains (technically, localhost:3001 and localhost:3002 are different), thus Cypress reloads the page completely and erases previous allure object.

I don't see any quick solution for this, as there is no clear way how to identify that Cypress decided to clear the state, so we could keep reporter object in some safe place, and then put into the new window state.

**Comment by PeaceDate** [comment]
@Shelex That make sence. I haven't figured out. Thank you a lot :)

**Comment by Shelex** [comment]
UPD:
What happens in case of changing domains:
- cypress clears the state
- creates new mocha runner
- replays events for suite - create new, start
- starts pending event for previous tests
- starts end event for previous tests

Without domain changing cypress fires "test pending" event only in case it is going to be skipped.
So in v2.35.1 I upgraded handler for "end test" event, and in case event was not skipped - set proper status.
However, as state was cleared, test object is empty, meaning there will be no steps attached - that is still to be investigated how to resolve.

**Comment by PeaceDate** [comment]
@Shelex HI, I checked and it works as you said. Thank you a lot, I'll continue monitoring this question for new changes and no steps attached issue. 

Have a good day and Take care of yourself : )

**Comment by PeaceDate** [comment]
@Shelex 
Isn't working with `Mocha.Context.skip()` when it add like Cypress custom command

```
Cypress.Commands.add('skipWhen', function (condition) {
    if (condition) {
        this.skip()
    }
})
```

Add such example in the [repo](https://github.com/PeaceDate/cypress-allure-plugin-bug) called `skipTestwithSkipWhen.cy.ts`

**Comment by Shelex** [comment]
@PeaceDate fixed in v2.35.2

**Comment by PeaceDate** [comment]
@Shelex Thank you, I've checked already. Everything is working fine 🔥
